### PR TITLE
Support for abbreviations

### PR DESCRIPTION
**Pull request by p1r473**
Add ability to show what alias you just used is aliased to (more like a "you used" opposite of "you should use" in case you forget you keep using aliases
Solves #129 
Split off from #131 

Can be disabled easily with
zstyle ':you-should-use:' you_should_use_alias_enabled false
zstyle ':you-should-use:*' you_should_use_abbreviation_enabled false

Also added something that I think makes sense
-local limit="${1:-9000000000000000}"
+local limit="${1:-$HISTSIZE}"

### CODE REVIEW

**Comment by AtifChy** [comment]
is this ready to be merged?

File: you-should-use.plugin.zsh [hunk 1]
```diff
@@ -111,7 +127,6 @@ You should use: ${PURPLE}\"%alias\"${NONE}"
     _write_ysu_buffer "$MESSAGE\n"
```
Comment (AtifChy) [inline you-should-use.plugin.zsh:127]: `_write_ysu_buffer` prints an empty line when `$MESSAGE` is empty

```
[[ -n $MESSAGE ]] && _write_ysu_buffer "$MESSAGE\n"
```
this should go here

Comment (AtifChy) [inline you-should-use.plugin.zsh:127]:
```suggestion
    [[ -n $MESSAGE ]] && _write_ysu_buffer "$MESSAGE\n"
```

File: you-should-use.plugin.zsh [hunk 2]
```diff
@@ -273,7 +360,16 @@ function enable_you_should_use() {
     add-zsh-hook preexec _check_global_aliases
     add-zsh-hook preexec _check_git_aliases
     add-zsh-hook precmd _flush_ysu_buffer
+    if zstyle -T ':you-should-use:*' you_should_use_abbreviation_enabled; then
+        add-zsh-hook preexec _check_abbrs
+    fi
 }
 
+zstyle ':you-should-use:*' you_should_use_alias_enabled true
+zstyle ':you-should-use:*' you_should_use_abbreviation_enabled false
```
Comment (AtifChy) [inline you-should-use.plugin.zsh:369]: for some reason tip for abbreviations refuse to show
`zstyle ':you-should-use:*' you_should_use_abbreviation_enabled false`
this is set in the plugin file

even if `zstyle ':you-should-use:*' you_should_use_abbreviation_enabled true` is set in `~/.zshrc` or ran interactively 
and `zstyle ':you-should-use:*' you_should_use_abbreviation_enabled` by default is false anyway so it this necessary?
```suggestion
# zstyle ':you-should-use:*' you_should_use_abbreviation_enabled false
```

**Comment by AtifChy** [comment]
Additionally, I've observed that it displays the abbreviation tip message, whether or not an abbreviation is used.

**Comment by p1r473** [comment]
I've accepted your suggestion @AtifChy this is ready to be merged @MichaelAquilina
I am not seeing any abbreviation tips if an abbreviation is not used

**Comment by p1r473** [comment]
Added natively into zsh-abbr https://github.com/MichaelAquilina/zsh-you-should-use/pull/131#issuecomment-2471377062

**Pull request closed without merging**